src/webgpu/api/operation/command_buffer/programmable/programmable_state_test.ts

```diff
@@ -9,31 +9,47 @@ interface BindGroupIndices {
 }
 
 export class ProgrammableStateTest extends GPUTest {
-  private commonBindGroupLayout: GPUBindGroupLayout | undefined;
-  private encoder: GPUCommandEncoder | null = null;
+  private commonBindGroupLayouts: Map<string, GPUBindGroupLayout> = new Map();
 
-  get bindGroupLayout(): GPUBindGroupLayout {
-    if (!this.commonBindGroupLayout) {
-      this.commonBindGroupLayout = this.device.createBindGroupLayout({
-        entries: [
-          {
-            binding: 0,
-            visibility: GPUShaderStage.COMPUTE | GPUShaderStage.FRAGMENT,
-            buffer: { type: 'storage' },
-          },
-        ],
-      });
+  private storageTypes: ('storage' | 'read-only-storage')[] = [
```

Comment:
I should probably comment these match the shader used in createBindingStatePipeline